### PR TITLE
[Fix][api] Fix build parameter error of sqlserver when create.

### PR DESCRIPTION
**Pull request by yh2388**
Fix jdbcUrl splicing error of SQLServer.

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java [hunk 1]
```diff
@@ -499,7 +499,13 @@ public String buildParameter(DbType type, String host,
...
         Map<String, Object> parameterMap = new LinkedHashMap<String, Object>(6);
-        String jdbcUrl = address + "/" + database;
+        String jdbcUrl;
+        if (DbType.SQLSERVER == type) {
+            jdbcUrl = address + ";databaseName=" + database;
+        } else {
+            jdbcUrl = address + "/" + database;
+        }
+
         if (Constants.ORACLE.equals(type.name())) {
```
Comment (gabrywu) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java:509]: Do you mind removing the following constants? I think using '==' is better, after all, the 'type' variable is Enum

    public static final String MYSQL = "MYSQL";
    public static final String POSTGRESQL = "POSTGRESQL";
    public static final String HIVE = "HIVE";
    public static final String SPARK = "SPARK";
    public static final String CLICKHOUSE = "CLICKHOUSE";
    public static final String ORACLE = "ORACLE";
    public static final String SQLSERVER = "SQLSERVER";
    public static final String DB2 = "DB2";
    public static final String PRESTO = "PRESTO";

**Review by gabrywu — DISMISSED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java (hunk 1 above)
Comment (qiaozhanwei) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java:509]: agree with @gabrywu idea . but this modify maybe involved not only DbType . suggest again submit a new PR

**Review by qiaozhanwei — APPROVED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]
Looking forward to your submission of a new pr to complete the changes required by @gabrywu

**Comment by yh2388** [comment]
> Looking forward to your submission of a new pr to complete the changes required by @gabrywu

okay.

**Pull request merged**